config_gen.py

```diff
@@ -162,6 +163,7 @@ def fake_build(project_dir, c_build_log_path, cxx_build_log_path, verbose, make_
     env_config = env.copy()
     env_config["YCM_CONFIG_GEN_CC_PASSTHROUGH"] = cc
     env_config["YCM_CONFIG_GEN_CXX_PASSTHROUGH"] = cxx
+    env_config["QMAKESPEC"] = "linux-clang"
```

Comment:
Makes sense. I think the value is right however:

```
florian@ws042:/usr/lib/x86_64-linux-gnu/qt5/mkspecs$ ls -d *clang*
linux-clang  linux-clang-libc++  macx-clang  macx-clang-32  macx-ios-clang
```

Note qmake is for the Qt project, which supports C++ only. The `-libc++` one would use `libc++` instead of `libstdc++` which is (I think) not what we want.